packages/sdk/src/types/ProductCollections.ts

```diff
@@ -0,0 +1,171 @@
+import { WithRequiredProperty } from './UtilityTypes'
+
+// Domain types
+// Product
+export interface ProductIdentity {
+	id?: string
+	object?: 'product'
+}
+
+export interface SkuIdentity {
+	id?: string
+	product_id?: string
+	object?: 'sku'
+}
+
+export interface ProductBase {
+	name?: string | null
+	price?: number | null
+	attributes?: string[]
+	metadata?: Record<string, unknown>
+}
+
+export interface SkuBase {
+	sku?: string | null
+	price?: number | null
+	attributes?: Record<string, unknown>
+	metadata?: Record<string, unknown>
+	product?: ProductInCollectionItem
+	image_url?: string | null
+}
+
+export interface ProductSaved {
+	created_at?: string
+	updated_at?: string | null
+	image_url?: string | null
+	object?: 'product'
+}
+
+export interface SkuSaved {
+	created_at?: string
+	updated_at?: string | null
+	image_url?: string | null
+	object?: 'sku'
+}
+
+export type Product = Required<SkuIdentity> | Required<ProductIdentity>
+
+export type ProductInCollection = Required<ProductIdentity> &
```

Comment:
This type is used in one place (inside `ProductCollectionsListProductsResponseBody`) - do we need to create it at all or we can just put definition directly inside `ProductCollectionsListProductsResponseBody`? Especially that it may be hard to figure out what's the difference between `ProductInCollection` and `ProductInCollectionItem`.